### PR TITLE
Added metrics for AMQP adapter

### PR DESCRIPTION
**Pull request by kaniyan**

### CODE REVIEW

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java [hunk 1]
```diff
@@ -354,7 +369,7 @@ private void handleSessionOpen(final ProtonConnection conn, final ProtonSession
      * @param con The connection to close.
      * @param res The client's close frame.
      */
-    private void handleRemoteConnectionClose(final ProtonConnection con, final AsyncResult<ProtonConnection> res) {
+    private void handleRemoteConnectionClose(final ProtonConnection con, final AsyncResult<ProtonConnection> res, final Device device) {
```
Comment (sophokles73) [inline adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java]: there doesn't seem to be a need for the device param

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java [hunk 2]
```diff
@@ -674,6 +689,7 @@ protected void onCommandReceived(final ProtonSender sender, final CommandContext
         });
         final Map<String, Object> items = new HashMap<>(4);
         items.put(Fields.EVENT, "command sent to device");
+        metrics.incrementCommandDeliveredToDevice(command.getTenant());
```
Comment (sophokles73) [inline adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java]: IMHO this needs to be moved to the block handling the the *accepted* outcome.

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java [hunk 3]
```diff
@@ -734,15 +750,27 @@ protected void onCommandReceived(final ProtonSender sender, final CommandContext
         })
         .map(downstreamDelivery -> {
             context.accept();
+            metrics.incrementProcessedMessages(context.getEndpoint(), context.getTenantId());
```
Comment (sophokles73) [inline adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java]: hmm, we also have the *hono.commands.response.delivered* metric which is supposed to track the number of responses being uploaded. However, it seems that we could as well simply use the *hono.messages.processed* metric for that purpose, we could then also record the response payload in *hono.messages.processed.payload* and get rid of *hono.commands.response.delivered* altogether.

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java (hunk 1 above)
Comment (kaniyan) [reply adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java]: I will remove it.

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java (hunk 3 above)
Comment (kaniyan) [reply adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java]: It sounds good to use the _hono.messages.processed.payload_ metric with type as  _command_ for the command response. This is already available and so I will remove the ` metrics.incrementProcessedMessages(context.getEndpoint(), context.getTenantId());`

Comment (kaniyan) [reply adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java]: It sounds good to use the _hono.messages.processed_ metrics for command responses too. I will remove the _hono.commands.response.delivered_ part.

Comment (kaniyan) [reply adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java]: :-) I will send a proposal to the mailing list.

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java (hunk 2 above)
Comment (kaniyan) [reply adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java]: The _accepted_ outcome is validated with _if/else_ blocks as below.

 `if (Accepted.class.isInstance(remoteState)) {
...
}`. 

I understood that the _incrementCommandDeliveredToDevice_ part should be moved there in consideration with the outcome. Is my understanding right? If not can you please explain where exactly do you mean?

**Comment by sophokles73** [comment]
obsolete with 13d21b34316e9b11e6fca3173d4e1ce393f2ebfa

**Pull request closed without merging**